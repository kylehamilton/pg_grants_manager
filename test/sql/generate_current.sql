BEGIN;
create user test_user;
create table test_table (col1 int4);
grant select on test_table to test_user;
select gm_generate_current();
select postgres, test_user from public.grants_manager where (schema_name, object_name, object_type) = ('public', 'test_table', 'TABLE');
ROLLBACK;