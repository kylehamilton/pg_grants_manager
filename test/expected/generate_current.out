BEGIN;
create user test_user;
create table test_table (col1 int4);
grant select on test_table to test_user;
select gm_generate_current();
NOTICE:  table "grants_manager_old" does not exist, skipping
CONTEXT:  SQL statement "drop table if exists grants_manager_old"
PL/pgSQL function gm_generate_current() line 11 at EXECUTE statement
NOTICE:  relation "grants_manager" does not exist, skipping
CONTEXT:  SQL statement "alter table if exists public.grants_manager rename to grants_manager_old"
PL/pgSQL function gm_generate_current() line 12 at EXECUTE statement
 gm_generate_current 
---------------------
 
(1 row)

select postgres, test_user from public.grants_manager where (schema_name, object_name, object_type) = ('public', 'test_table', 'TABLE');
 postgres | test_user 
----------+-----------
          | {r}
(1 row)

ROLLBACK;
